pkg/arvo/app/btc-node-hook.hoon

```diff
@@ -0,0 +1,354 @@
+          (request-to-rpc:btc-rpc:lib act)
+        =/  req=request:http
+          :*  %'POST'
+              (endpoint-url:bc act)
+              headers
+              =,  html
+              %-  some
+                %-  as-octt:mimes
+                  (en-json (request-to-json:rpc:jstd body))
+          ==
+        =/  out  *outbound-config:iris
+        :_  this
+        [%pass /[(scot %da now.bowl)] %arvo %i %request req out]~
+      ::
+      ++  hook-command
+        |=  comm=btc-node-hook-command
+        ^-  (quip card _this)
+        ?+    -.comm  ~|  [%unsupported-hook-command -.comm]  !!
+            %credentials
+          :_  this(endpoint url.comm, headers heads.comm)
+          [%pass / %arvo %d %flog [%text "credentials updated..."]]~
+        ==
+      --
+    ::
+    ++  on-watch  on-watch:def
+    ++  on-leave  on-leave:def
+    ++  on-peek   on-peek:def
+    ++  on-agent  on-agent:def
+    ++  on-arvo
+      |=  [=wire =sign-arvo]
+      ^-  (quip card _this)
+      =*  response  client-response.sign-arvo
+      =^  cards  state
+        ?+    +<.sign-arvo    (on-arvo:def wire sign-arvo)
+            %http-response    (http-response:bc wire response)
+        ==
+      [cards this]
+    ::
+    ++  on-fail   on-fail:def
+    --
+::
+|_  =bowl:gall
+::
+++  httr-to-rpc-response
+  |=  hit=httr:eyre
+  ^-  response:rpc:jstd
+  ~|  hit
+  ?.  ?=($2 (div p.hit 100))
+    fail+hit
+  =/  a=json  (need (de-json:html q:(need r.hit)))
+  =,  dejs-soft:format
+  ^-  response:rpc:jstd
+  =;  dere
+    =+  res=((ar dere) a)
+    ?~  res  (need (dere a))
+    [%batch u.res]
+  |=  a=json
+  ^-  (unit response:rpc:jstd)
+  =/  res=(unit [@t json])
+    ::  TODO  breaks when no id present
+    ::
+    ((ot id+so result+some ~) a)
+    ::  TODO: Modify to use dejs instead of dejs-soft
+    ::
+    :: %.  a
+    :: =-  (ou -)
+    :: :~  ['id' (uf ~ (mu so))]
+    ::     ['result' some]
+    :: ==
+  ?^  res  `[%result u.res]
+  ~|  a
+  :+  ~  %error  %-  need
+  ((ot id+so error+(ot code+no message+so ~) ~) a)
+::
+++  http-response
+  |=  [=wire response=client-response:iris]
+  ?.  ?=(%finished -.response)
+    [~ state]
+  =*  status    status-code.response-header.response
+  =/  rpc-resp=response:rpc:jstd
+    %-  httr-to-rpc-response
+      %+  to-httr:iris
+        response-header.response
+      full-file.response
+  ::  Only (FOR NOW) parse successful responses
+  ::
+  ?.  =(%result -.rpc-resp)
+    ::  TODO: parse error response properly
+    ::
+    ~&  [%error +.rpc-resp]
+    [~ state]
+  %-  handle-btc-response
+    (parse-response:btc-rpc:lib rpc-resp)
+::
+++  handle-btc-response
+  |=  btc-resp=btc-node-hook-response
+  ^-  (quip card _state)
+  :_  state
+  ^-  (list card)
+  ;:  weld
+      ::  Print term for each succesful RPC request (used by :ph)
+      ::
+      ^-  (list card)
+      [%pass / %arvo %d %flog [%text (trip -.btc-resp)]]~
+      ?+    -.btc-resp  ::  ~|  [%unsupported-response -.btc-resp]  !!
```

Comment:
Just pull this big ol' `?+` out of the `weld` call. You could do `%+  weld  [%pass ...]~` I think, with the `?+` on the next line. Perhaps the `%flog`ing and actual response handling should be in separate arms in the first place.

Reply:
I'm going to remove the flogging since its purpose was to detect successful calls in pH, but I ended up not using it.